Super easy login

axelson opened the pull request:
When I was at NervesConf I noticed that a few people had trouble realizing that `ssh nerves.local` would not work and instead they needed to do `ssh livebook@nerves.local`. This is an attempt at fixing that (or removing the restriction completely!)

This is probably not quite how we want to do it. But there are two options for improvement and I figured I'd just show them both on one PR.

Set `:pwdfun` to control what the valid passwords are, the simplest implementation of this is to just return `true` (which is currently in this PR), this will allow an ssh connect to succeed with **ANY** username and password combo. Very easy. Also very insecure, but is it that much different than stating the correct username and password at the SSH prompt? Another way to do this would be to allow any username, but only the password "nerves".

The other improvement that could be made (and I think should be) is to warn the user if they are using an incorrect password, along with showing the actual node name instead of a hard-coded `nerves.local`. This is accomplished by passing a function to `:auth_method_kb_interactive_data`.

Here's how that looks (note that nerves-517f.local is my rpi0 hostname, it will be different when you run this):

```
jason@jdesktop ~/d/f/nerves_livebook (frank-main)> ssh nerves.local
Nerves Livebook
https://github.com/livebook-dev/nerves_livebook

Wrong username, should use username 'livebook' (received jason)

ssh livebook@nerves-517f.local # Use password "nerves"

(jason@livebook.local) Password:
```

```
jason@jdesktop ~/d/f/nerves_livebook (easy-login)> ssh nerves@livebook.local
Nerves Livebook
https://github.com/livebook-dev/nerves_livebook

Use password "nerves"

(livebook@livebook.local) Password:
```

If we want to move forward with this approach there's some consistency fixes to do in those messages.

fhunleth commented:
Thanks!

I've forgotten to set the user so many times that I definitely appreciate this PR. I had been dragging my feet on fixing this mostly since I wanted to provide a way to set a password for both the UI and ssh rather than having the hardcoded one. 

I didn't know that you could just pass a function. Perhaps ignoring the username is the right thing to do in the interim. I agree with the argument that there's not any security here. Something just feels like crossing a line when every username/password works.

axelson commented:
> I didn't know that you could just pass a function. Perhaps ignoring the username is the right thing to do in the interim. I agree with the argument that there's not any security here. Something just feels like crossing a line when every username/password works.

Yeah I think that makes sense as a line not to cross. I've updated the PR to match that.

And I've updated the PR to read the password from the Livebook configuration (which is a little bit hacky, but it falls back to `"nerves"` which helps somewhat).

I've only kept a small update to the ssh message, it now uses the actual system hostname instead of a hardcoded nerves.local.

fhunleth commented:
Thanks for the updates. Looks good!
